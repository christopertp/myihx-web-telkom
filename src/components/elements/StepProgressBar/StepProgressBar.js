import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function StepProgressBar(props) {
  const { className, steps, complete } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');

  return (
    <ul className={customClass}>
      {Array(steps)
        .fill(steps)
        .map((val, idx) => (
          <Fragment key={idx}>
            <li className={idx >= steps - complete ? styles.complete : ''} key={idx} />
            <span />
          </Fragment>
        ))}
    </ul>
  );
}

StepProgressBar.defaultProps = {
  className: '',
  complete: 3,
  steps: 3,
};

StepProgressBar.propTypes = {
  className: PropTypes.string,
  complete: PropTypes.number,
  steps: PropTypes.number,
};
