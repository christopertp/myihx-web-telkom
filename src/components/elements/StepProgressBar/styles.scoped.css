.root {
  align-items: center;
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
}

.root > li {
  background-color: #cccdce;
  border-radius: 50%;
  color: #7d7d7d;
  height: 1.25rem;
  width: 1.25rem;
  z-index: 2;
}

.root > li + span {
  background-color: #e5e6e6;
  height: 0.5rem;
  flex: 1;
  margin: 0 -0.25rem;
  z-index: 1;
}

.root > li + span:last-child {
  display: none;
}

.root > li.complete {
  background-image: url('/assets/ic_done.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: 110%;
  height: 1.25rem;
  width: 1.25rem;
}

.root > li.complete + span {
  background-color: #18af7c;
  color: #18af7c;
}
