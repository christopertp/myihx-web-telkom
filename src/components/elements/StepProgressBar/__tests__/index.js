import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import StepProgressBar from '../StepProgressBar';

describe('src/components/elements/StepProgressBar', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<StepProgressBar />);
    expect(tree).toMatchSnapshot();
  });

  test('render with completed', () => {
    const result1 = StepProgressBar({  ...StepProgressBar.defaultProps, steps: 5, complete: 5 });
    expect(result1.props.children[4].props.children[0].props.className).toBe(undefined);

    const result2 = StepProgressBar({ ...StepProgressBar.defaultProps, steps:5, complete:0 });
    expect(result2.props.children[4].props.children[0].props.className).toBe('');
  });
});
