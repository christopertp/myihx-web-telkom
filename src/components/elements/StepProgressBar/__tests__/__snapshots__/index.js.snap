// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/StepProgressBar render 1`] = `
<ul
  className=""
>
  <React.Fragment>
    <li />
    <span />
  </React.Fragment>
  <React.Fragment>
    <li />
    <span />
  </React.Fragment>
  <React.Fragment>
    <li />
    <span />
  </React.Fragment>
</ul>
`;
